cypress-io/cypress
# fix: run-all-specs opens in new tab rather than new browser

- Closes #24919
### User facing changelog
Run All Specs will open in a new tab rather than close and reopen the browser

### Additional details
Following the same [run.ts](https://github.com/cypress-io/cypress/blob/294e915b9f0f84b95eb683df111773a1220283a1/packages/server/lib/modes/run.ts#L642) logic for retaining the browser and switching out the tab with a fresh one.

### Steps to test
Choose a project that has Cypress setup and add `e2e.experimentalRunAllSpecs = true` to the config. Check out this branch, run `yarn` and the `yarn cypress:open`. Open up a browser and verify that, after clicking "Run all specs", the browser isn't closed and the current tab is replaced with a new one.

Electron still spawns a new window since there is no "New Tab" in Electron.

Not sure the best way to test this change, since any tests around `launchProject` require opening and closing browsers.

### How has the user experience changed?

https://user-images.githubusercontent.com/25158820/206529057-a489559c-9538-4887-a2e9-207176fa7dcc.mov

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
